enhancements/sig-policy/28-selective-policy-enforcment/README.md

```diff
@@ -0,0 +1,624 @@
+1. Create set "R" of all clusters that are placed with placement
+   bindings with subFilter: true and
+   remediationActionOverride.remediationAction: enforce.
+1. Create set "S" which is the intersection of set "P" and set "R".
+1. The clusters which have the remediationAction overridden is the
+   union of set "Q" and set "S".
+
+### Examples
+In the following examples an inform Policy is bound to
+`placementrule-initial` which selects clusters [ "A", "B", "C", "D"
+]. Additional placement rules are defined and used in the examples
+below:
++ `placementrule-sub` selects clusters [ "A", "B" ]
++ `placementrule-sub-2` selects clusters [ "E", "F" ]
++ `placementrule-extended` selects clusters [ "A", "B", "E", "F" ]
+
+#### Example 1: Simple selection
+```
+kind: PlacementBinding
+placementRef:
+  apiGroup: apps.open-cluster-management.io
+  kind: PlacementRule
+  name: placementrule-sub
+remediationActionOverride:
+    remediationAction: enforce
+subjects:
+- apiGroup: policy.open-cluster-management.io
+  kind: Policy
+  name: test-policy-1
+```
+
+The initial policy binds to clusters A, B, C, and D. The
+placementrule-sub selects clusters A and B. Based on the override the
+policy is switched to `enforce` mode for clusters A and B. The policy
+remains in `inform` mode for clusters C and D.
+
+#### Example 2: No sub filtering
+```
+kind: PlacementBinding
+placementRef:
+  apiGroup: apps.open-cluster-management.io
+  kind: PlacementRule
+  name: placementrule-extended
+remediationActionOverride:
+    remediationAction: enforce
+    subFilter: false
+subjects:
+- apiGroup: policy.open-cluster-management.io
+  kind: Policy
+  name: test-policy-1
+```
+
+The `subFilter` field is false so all clusters in all placementrules
+which are bound to the policy are considered. In this case the bound
+PlacementRules are placementrule-initial and placementrule-extended
+(in this binding) so that clusters A, B, C, D, E, and F are all
+considered.
+
+This PlacementBinding includes placmentrule-extended which includes A,
+B, E, and F. This means clusters A, B, E and F will be set to enforce
+mode. Clusters B and C remain in inform mode.
+
+#### Example 3: With sub filtering
+```
+kind: PlacementBinding
+placementRef:
+  apiGroup: apps.open-cluster-management.io
+  kind: PlacementRule
+  name: placementrule-extended
+remediationActionOverride:
+    remediationAction: enforce
+    subFilter: true
+subjects:
+- apiGroup: policy.open-cluster-management.io
+  kind: Policy
+  name: test-policy-1
+```
+
+The `subFilter` field is true in this example so only clusters in
+placementrules with `subFilter` that is false are considered. In this
+case that is only the placementrule-initial with clusters A, B, C, and
+D. The PlacementRule in this CR is `placementrule-extended` which
+includes A, B, E, and F.
+
+The set of clusters set to enforce mode is the intersection of these
+two sets. Clusters A and B are set to enforce mode.
```

Comment:
So E and F will still be bound without action override? Not sure in what use case this kind of mixed binding behavior would be useful. For our main use case from TALM, it would make more sense to keep the subfilter at the binding level so that the new binding only does action override to already bound clusters and no other clusters get impacted.

Reply:
Yes, E and F are bound only in this override placement binding which is created to apply the policy to a subset of the original bound clusters. When the user sets subFilter to "true" they are saying "I don't want this binding/override to add any new clusters to the set that the policy applies to, I'm just trying to enforce it to some of the originally bound ones". The subFilter is a safety mechanism to avoid accidentally adding more clusters into the set.